From: Adrian Bunk <bunk@debian.org>
Date: Tue, 3 Jul 2018 10:03:36 +0100
Subject: Use AtomicOperations-sparc.h instead of AtomicOperations-none.h

Bug-Debian: https://bugs.debian.org/878284
---
 js/src/jit/AtomicOperations.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/src/jit/AtomicOperations.h b/js/src/jit/AtomicOperations.h
index 42aee72..f720cb4 100644
--- a/js/src/jit/AtomicOperations.h
+++ b/js/src/jit/AtomicOperations.h
@@ -342,7 +342,8 @@ AtomicOperations::isLockfree(int32_t size)
 # elif defined(__aarch64__)
 #  include "jit/arm64/AtomicOperations-arm64.h"
 # else
-#  include "jit/none/AtomicOperations-none.h" // These MOZ_CRASH() always
+  // The sparc file is more or less generic
+#  include "jit/none/AtomicOperations-sparc.h"
 # endif
 #elif defined(JS_CODEGEN_X86) || defined(JS_CODEGEN_X64)
 # include "jit/x86-shared/AtomicOperations-x86-shared.h"
