--- mozjs52-52.3.1/js/src/jit/ProcessExecutableMemory.cpp.orig	2017-08-08 12:25:50.000000000 +0200
+++ mozjs52-52.3.1/js/src/jit/ProcessExecutableMemory.cpp	2018-05-08 12:36:58.638597620 +0200
@@ -290,9 +290,20 @@
     void* randomAddr = ComputeRandomAllocationAddress();
     void* p = MozTaggedAnonymousMmap(randomAddr, bytes, PROT_NONE, MAP_PRIVATE | MAP_ANON,
                                      -1, 0, "js-executable-memory");
+
+#ifndef __ia64__
     if (p == MAP_FAILED)
         return nullptr;
+#else
+    if (p == MAP_FAILED) {
+        // the comment above appears to be incorrect on ia64, so retry without the hint
+        p = MozTaggedAnonymousMmap(NULL, bytes, PROT_NONE, MAP_PRIVATE | MAP_ANON,
+                                     -1, 0, "js-executable-memory");
+        if (p == MAP_FAILED)
+            return nullptr;
+    }
     return p;
+#endif
 }
 
 static void
